src/dtemplate.d

```diff
@@ -8715,6 +8715,25 @@ public:
         return 0;
     }
 
+    override int apply2(Dsymbol_apply_ft_t fp, void* param)
```

Comment:
My problem with this PR is the addition of `apply2` which is the same only different from `apply`. I don't know why one should work on TemplateMixins and the other not. The fact that there's a difference needs to be explained, it looks like a bug.